src/utils/analyzer/FunctionCallAnalyzer.ts

```diff
@@ -284,13 +302,27 @@ export class FunctionCallAnalyzer {
     private readonly updateInputResult = () => {
         const ff = this.functionFragment.value
         const inputArgs = this.inputArgsOnly.value
+        const contractId = this.contractAnalyzer.contractId.value
+        const functionHash = this.functionHash.value
+
         if (ff !== null && inputArgs !== null) {
             try {
                 this.inputResult.value = ethers.AbiCoder.defaultAbiCoder().decode(ff.inputs, inputArgs)
                 this.inputDecodingFailure.value = null
-            } catch (failure) {
-                this.inputResult.value = null
-                this.inputDecodingFailure.value = failure
+            } catch(failure) {
+                const f = failure as ethers.EthersError
+                // please refer to the ticket below for more information on this logic for redirectForToken(address,bytes) method on HTS System Contract 
+                // https://github.com/hashgraph/hedera-mirror-node-explorer/issues/921
+                if (
+                    (f.code === "BUFFER_OVERRUN" || f.code === "INVALID_ARGUMENT") &&
+                    contractId === HTS_PRECOMPILE_CONTRACT_ID &&
+                    functionHash === REDIRECT_FOR_TOKEN_FUNCTION_SIGHASH
+                ) {
+                    this.inputResult.value = decodeRedirectForTokenInput(inputArgs)
+                } else {
+                    this.inputResult.value = null
+                    this.inputDecodingFailure.value = failure
+                }
```

Comment:
Here too we can re-use `isRedirectForToken()` and have a straight if-then-else flow:

```
            try {
                if (contractId !== null && functionHash !==  null) {
                    if (isRedirectToken(contractId, functionHash)) {
                        this.inputResult.value = decodeRedirectForTokenInput(inputArgs)
                        this.inputDecodingFailure.value = null
                    } else {
                        this.inputResult.value = ethers.AbiCoder.defaultAbiCoder().decode(ff.inputs, inputArgs)
                        this.inputDecodingFailure.value = null
                    }
                } else {
                    this.inputResult.value = null
                    this.inputDecodingFailure.value = null
                }
            } catch(failure) {
                this.inputResult.value = null
                this.inputDecodingFailure.value = failure
            }

```